scripts/sentry-xcodebuild.sh

```diff
@@ -57,63 +63,122 @@ fi
 
 case $COMMAND in
 "build")
+    RUN_ANALYZE=false
     RUN_BUILD=true
     RUN_BUILD_FOR_TESTING=false
     RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=false
     ;;
 "build-for-testing")
+    RUN_ANALYZE=false
     RUN_BUILD=false
     RUN_BUILD_FOR_TESTING=true
     RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=false
     ;;
 "test-without-building")
+    RUN_ANALYZE=false
     RUN_BUILD=false
     RUN_BUILD_FOR_TESTING=false
     RUN_TEST_WITHOUT_BUILDING=true
+    RUN_TEST_WITH_TSAN=false
+    ;;
+"analyze")
+    RUN_ANALYZE=true
+    RUN_BUILD=false
+    RUN_BUILD_FOR_TESTING=false
+    RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=false
+    ;;
+"tsan")
+    RUN_ANALYZE=false
+    RUN_BUILD=false
+    RUN_BUILD_FOR_TESTING=false
+    RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=true
     ;;
 *)
+    RUN_ANALYZE=false
     RUN_BUILD=false
     RUN_BUILD_FOR_TESTING=true
     RUN_TEST_WITHOUT_BUILDING=true
+    RUN_TEST_WITH_TSAN=false
     ;;
 esac
 
-if [ "$CI" == true ]; then
+if [ "${CI-false}" == true ]; then
```

Comment:
`l`: why the fallback to false? isn't `"" == true` also falsy?

Reply:
the script fails here if you run it locally if your machine doesn't define the `CI` environment variable at all

Reply:
Why does it fail when a variable is not set? shouldn't it just evaluate to an empty string?

Reply:
hm, after trying another minimal example it does. i was getting an unbound variable error at some point. i'll fix this.